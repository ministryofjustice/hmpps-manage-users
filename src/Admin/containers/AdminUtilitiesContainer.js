import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { setLoaded } from '../../redux/actions'
import Page from '../../Components/Page'
import { userType, configType } from '../../types'
import MessageBar from '../../MessageBar'
import { AdminUtilities, AdminUtility } from './AdminUtilitiesContainer.styles'

export class AdminUtilitiesContainer extends Component {
  componentWillMount() {
    const { setLoadedDispatch } = this.props

    setLoadedDispatch(true)
  }

  render() {
    const { user } = this.props
    const hasMaintainNomisRolesAccess = user && (user.maintainAccess || user.maintainAccessAdmin)
    const hasMaintainAuthUsers = user && (user.maintainAuthUsers || user.groupManager)

    return (
      <Page title="Manage user accounts">
        <MessageBar {...this.props} />
        <AdminUtilities>
          {!hasMaintainNomisRolesAccess && !hasMaintainNomisRolesAccess && !hasMaintainAuthUsers && (
            <div>
              <p>There are no admin functions associated with your account.</p>
            </div>
          )}
          {hasMaintainNomisRolesAccess && (
            <AdminUtility>
              <Link id="maintain_roles_link" title="Manage user roles" className="link" to="/maintain-roles">
                Manage user roles
              </Link>
            </AdminUtility>
          )}
          {hasMaintainAuthUsers && (
            <AdminUtility>
              <Link
                id="maintain_auth_users_link"
                title="Maintain external users"
                className="link"
                to="/maintain-external-users"
              >
                Maintain external users
              </Link>
              <div>Maintain users that do not exist in NOMIS, only in auth.</div>
            </AdminUtility>
          )}
          {hasMaintainAuthUsers && (
            <AdminUtility>
              <Link
                id="create_auth_user_link"
                title="Create external users"
                className="link"
                to="/create-external-user"
              >
                Create external users
              </Link>
              <div>Create user in auth (not in NOMIS).</div>
            </AdminUtility>
          )}
        </AdminUtilities>
      </Page>
    )
  }
}

AdminUtilitiesContainer.propTypes = {
  user: userType.isRequired,
  config: configType.isRequired,
  setLoadedDispatch: PropTypes.func.isRequired,
}

const mapStateToProps = (state) => ({
  user: state.app.user,
  config: state.app.config,
})

const mapDispatchToProps = (dispatch) => ({
  setLoadedDispatch: (status) => dispatch(setLoaded(status)),
})

export default connect(mapStateToProps, mapDispatchToProps)(AdminUtilitiesContainer)
