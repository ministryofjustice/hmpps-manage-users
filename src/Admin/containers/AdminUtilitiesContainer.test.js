import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'
import { AdminUtilitiesContainer } from './AdminUtilitiesContainer'

const props = {
  setLoadedDispatch: jest.fn(),
  user: {
    maintainAccessAdmin: false,
    activeCaseLoadId: '',
    caseLoadOptions: [],
    expiredFlag: false,
    firstName: 'Test',
    lastName: 'User',
    lockedFlag: false,
    maintainAccess: false,
    staffId: 1,
    username: 'TestUser',
  },
  config: {
    mailTo: 'email@test.com',
    notmEndpointUrl: '//notm.url',
  },
  message: '',
}

describe('<AdminUtilitiesContainer />', () => {
  it('render with a message if there are no admin rights', () => {
    const wrapper = shallow(<AdminUtilitiesContainer {...props} />)

    expect(wrapper.find('p').text()).toEqual('There are no admin functions associated with your account.')
  })

  describe('render links', () => {
    it('should render a Link to Maintain Roles Admin section if user has maintain access role', () => {
      props.user.maintainAccess = true
      const wrapper = shallow(<AdminUtilitiesContainer {...props} />)
      const manageRolesLink = wrapper.find('Link').find({ to: '/maintain-roles' })

      expect(manageRolesLink.prop('children')).toEqual('Manage NOMIS user roles')
    })

    it('should render a Link to Maintain auth users if user has maintain auth user roles', () => {
      props.user.maintainAuthUsers = true
      const wrapper = shallow(<AdminUtilitiesContainer {...props} />)
      const maintainUsersLink = wrapper.find('Link').find({ to: '/maintain-auth-users' })

      expect(maintainUsersLink.prop('children')).toEqual('Manage auth users')
    })

    it('should render a Link to Maintain auth users if user has group manager role', () => {
      props.user.groupManager = true
      const wrapper = shallow(<AdminUtilitiesContainer {...props} />)
      const maintainUsersLink = wrapper.find('Link').find({ to: '/maintain-auth-users' })

      expect(maintainUsersLink.prop('children')).toEqual('Manage auth users')
    })

    it('should render a Link to create auth users if user has maintain auth user roles', () => {
      props.user.maintainAuthUsers = true
      const wrapper = shallow(<AdminUtilitiesContainer {...props} />)
      const maintainUsersLink = wrapper.find('Link').find({ to: '/create-auth-user' })

      expect(maintainUsersLink.prop('children')).toEqual('Create auth user')
    })
  })

  it('should render the Maintain HMPPS Users page', () => {
    props.user.maintainAccessAdmin = true
    props.user.migration = true
    props.user.maintainAuthUsers = true
    const store = { subscribe: jest.fn(), dispatch: jest.fn(), getState: jest.fn(), setState: jest.fn() }
    store.getState.mockReturnValue({ app: { error: '', loaded: true } })

    const rendered = renderer.create(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/']}>
          <AdminUtilitiesContainer {...props} />
        </MemoryRouter>
      </Provider>
    )

    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
