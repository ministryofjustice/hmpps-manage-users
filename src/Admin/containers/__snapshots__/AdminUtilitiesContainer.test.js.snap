// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminUtilitiesContainer /> should render the Maintain HMPPS Users page 1`] = `
Array [
  <div
    className="sc-dnqmqq covUjf"
  >
    <ul
      className="sc-iwsKbI cWIuJK"
    >
      <li
        className="sc-gZMcBi cRHLOy"
      >
        <a
          data-qa="breadcrumb-home-page-link"
          href=""
        >
          Home
        </a>
      </li>
      <li
        className="sc-gZMcBi cRHLOy"
      >
        <span>
          Maintain HMPPS Users
        </span>
      </li>
    </ul>
  </div>,
  <div
    className="sc-htoDjs bCKyuL"
  >
    
    <h1
      className="sc-bdVaJa dSUtdh"
      size="LARGE"
    >
      Maintain HMPPS Users
    </h1>
    <div
      className="page-content"
    >
      <div
        className="sc-gqjmRU cthAvw"
      >
        <div
          className="sc-VigVT dUvdCr"
        >
          <a
            className="link"
            href="/maintain-roles"
            id="maintain_roles_link"
            onClick={[Function]}
            title="Manage NOMIS user roles"
          >
            Manage NOMIS user roles
          </a>
          <div>
            Add and remove NOMIS user roles.
          </div>
        </div>
        <div
          className="sc-VigVT dUvdCr"
        >
          <a
            className="link"
            href="/maintain-auth-users"
            id="maintain_auth_users_link"
            onClick={[Function]}
            title="Manage auth users"
          >
            Manage auth users
          </a>
          <div>
            Maintain users that do not exist in NOMIS, only in auth.
          </div>
        </div>
        <div
          className="sc-VigVT dUvdCr"
        >
          <a
            className="link"
            href="/create-auth-user"
            id="create_auth_user_link"
            onClick={[Function]}
            title="Create auth user"
          >
            Create auth user
          </a>
          <div>
            Create user in auth (not in NOMIS).
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
